testing.rst

```diff
@@ -139,249 +170,313 @@ tests as far as PHPUnit is concerned, but they have a very specific workflow:
-        // HTTP status numbers are available as constants too:
-        // e.g. 201 === Symfony\Component\HttpFoundation\Response::HTTP_CREATED
-        // equivalent to:
-        $this->assertEquals(201, $client->getResponse()->getStatusCode());
-
-        // asserts that the response is a redirect to /demo/contact
-        $this->assertResponseRedirects('/demo/contact');
-        // equivalent to:
-        $this->assertTrue($client->getResponse()->isRedirect('/demo/contact'));
-        // ...or check that the response is a redirect to any URL
-        $this->assertResponseRedirects();
+    // src/DataFixtures/ProductFixture.php
+    namespace App\DataFixtures;
 
-.. versionadded:: 4.3
+    use App\Entity\Product;
+    use Doctrine\Bundle\FixturesBundle\Fixture;
+    use Doctrine\Persistence\ObjectManager;
 
-    The ``assertResponseHeaderSame()``, ``assertResponseIsSuccessful()``,
-    ``assertResponseStatusCodeSame()``, ``assertResponseRedirects()`` and other
-    related methods were introduced in Symfony 4.3.
+    class ProductFixture extends Fixture
+    {
+        public function load(ObjectManager $manager)
+        {
+            $product = new Product();
+            $product->setName('Priceless widget');
+            $product->setPrice(14.50);
+            $product->setDescription('Ok, I guess it *does* have a price');
+            $manager->persist($product);
 
-.. _testing-data-providers:
+            // add more products
 
-Testing against Different Sets of Data
---------------------------------------
+            $manager->flush();
+        }
+    }
 
-It's common to have to execute the same test against different sets of data to
-check the multiple conditions code must handle. This is solved with PHPUnit's
-`data providers`_, which work both for unit and functional tests.
+Empty the database and reload *all* the fixture classes with:
 
-First, add one or more arguments to your test method and use them inside the
-test code. Then, define another method which returns a nested array with the
-arguments to use on each test run. Lastly, add the ``@dataProvider`` annotation
-to associate both methods::
+.. code-block:: terminal
 
-    /**
-     * @dataProvider provideUrls
-     */
-    public function testPageIsSuccessful($url)
-    {
-        $client = self::createClient();
-        $client->request('GET', $url);
+    $ php bin/console doctrine:fixtures:load
 
-        $this->assertTrue($client->getResponse()->isSuccessful());
-    }
+For more information, read the `DoctrineFixturesBundle documentation`_.
+
+Write Your First Functional Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Functional tests are PHP files that typically live in the ``tests/Controller``
+directory of your application. If you want to test the pages handled by your
+``PostController`` class, start by creating a new ``PostControllerTest.php``
+file that extends a special ``WebTestCase`` class.
+
+As an example, a test could look like this::
```

Comment:
what about referencing `make:functional-test` and `make:unit-test` from the https://github.com/symfony/maker-bundle?

can be a good bootstraping code, like for fixtures above :)

Reply:
For the record, these commands are now deprecated and a new `make:test` command has been introduced: https://github.com/symfony/maker-bundle/blob/main/src/Maker/MakeTest.php#L38-L42

IMHO the docs should be consistent with this command, and explain all `TestCase` types that are available.

Reply:
Yes, agreed. For now, I've updated the "Application Tests" section to use the new `make:test` command. Let's rework this thing in a follow-up PR.